crates/handler/src/precompile_provider.rs

```diff
@@ -98,9 +98,6 @@ impl<CTX: ContextTr> PrecompileProvider<CTX> for EthPrecompiles {
         _is_static: bool,
         gas_limit: u64,
     ) -> Result<Option<InterpreterResult>, String> {
-        let Some(precompile) = self.precompiles.get(address) else {
-            return Ok(None);
-        };
```

Comment:
We now check for this in `self.precompiles.execute` -- will revert this and just deref the fn pointer, then explicitly pass in the dyn trait